File: python/algorithms/insertion.py
from util import swap_int

def insertion_sort(arr):
    n = len(arr)
    if n < 2:
        return
    for i in range(1, n):
        j = i
        while j > 0 and arr[j - 1] > arr[j]:
            swap_int(arr, j, j - 1)
            j -= 1

File: cpp/algorithms/insertion.cpp
#include "insertion.hpp"
#include "../util.hpp"

void insertion_sort(std::vector<int>& arr) {
    if (arr.size() < 2) return;

    for (size_t i = 1; i < arr.size(); i++) {
        size_t j = i;
        while (j > 0 && arr[j - 1] > arr[j]) {
            swap_int(arr[j], arr[j - 1]);
            j--;
        }
    }
}
